### ISSUE TITLE
Default value for component props

### ISSUE THREAD

**Issue by za-creature** [yyx990803/vue#973]
Would be nice to have:

```javascript
Vue.component("test", {
    "props": [{
        "name": "visible"
        "type": Boolean
        "default": true
    }]
})
```

Apologies if there are easy ways to implement this (i.e. not by setting manually in the constructor)

**Comment by yyx990803** [comment]
Props are just data, so what you return in `data` will become the default if the parent didn't provide it:

``` js
props: [{
  name: 'visible',
  type: Boolean
}],
data: function () {
  return {
    visible: true
  }
}
```

Although I agree that having the default directly in `props` seems better.

**Comment by young-steveo** [comment]
Usually if a key is not defined in `data` it is not observed; having it defined in `props` as a default value without also defining it in `data` breaks that assumption.

**Comment by mark-hahn** [comment]
> without also defining it in data breaks that assumption.

Isn't everything in props independent of data whether there is a default or
not?

**Comment by Nirazul** [comment]
I'd love to see that, too. If a prop has its property set to `required: false`, it'd be natural to assume that a default value can be provided, that doesn't necessarily needs to come from the parent.

**Comment by young-steveo** [comment]
@mark-hahn hmm... yeah, maybe I'm wrong.

**Comment by yyx990803** [comment]
Closed via 6afe4c5af5517183f636df4d450fbc508a1f5509
